language.md

```diff
@@ -625,3 +625,125 @@ seen on the status page, or:
+during a `Group`, `Join`, `Smash` or `Let` operation if not preserved. Once the
+input is manipulated by a `Group`, `Join`, `Smash` or `Let` operation, it is
+not possible to track what input was used. So, references are only considered
+from the start of the olive until the first manipulation operation. If it is a
+`Run` olive, with no manipulation, the `With` arguments are included.
+
+The collection of included variables is considered over the whole scope. For
+instance:
+
+   Run x
+     Where "project" In signable_names # This is true even though project is
+                                       # referenced after this check
+     Where project ~ /N.*/
+     With { project = project }
+
+Since there are no `Group`, `Join`, `Smash` or `Let` clauses, the entire olive
+is in scope. The signable variable `project` is used twice: once in the `Where`
+clause and once in the arguments to the action. Therefore `project` is one of
+the referenced variables and appears in `signable_names` and its value will be
+in `sha1_signature`. Order does not matter: although `project` is referenced
+after `signature_names` is used, it is still present because it is used in that
+scope.
+
+This behaviour is necessary to ensure that `sha1_signature` returns the same
+value in all parts of the program.
+
+#### Example
+As an example, let's suppose we want to run a variant caller. It will take a
+list of genome alignments for different tumour/normal pairs in a patient and
+produce variant information. Whether the tissue is reference or normal is
+`tissue_type`. Now, the `tissue_type` is associated with each genome alignment
+(BAM) file, but if the tissue type is changed, the BAM file is not, so the
+action that produces the BAM will not be re-run. However, it affects the
+variant caller, so it should be triggered to rerun even though the input BAM is
+not changed.
+
+    Input gsi_std;
+    
+    Run variant_caller
+      Where metatype == "application/bam"
+      Max timestamp By donor, tissue_type
+      Smash
+          reference = path Where tissue_type == "R",
+          reference_signature = sha1_signature Where tissue_type == "R",
+          tumour = path Where tissue_type == "T",
+          tumour_signature = sha1_signature Where tissue_type == "T"
+        By donor
+      With {
+        input_signaturees = [reference_signature, tumour_signature],
+        reference_file = reference,
+        tumour_file = tumour
+      }
+
+By saving the signatures as part of the action, we save the input information.
+In this example, the signature will save `donor`, and `tissue_type`.  That
+means if there was a sample swap and both tissues belong to a different donor,
+the signatures will change and the action will be different and, therefore,
+re-run, even though the donor is not directly included in the parameters to the
+action.  Not all values are included. The `metatype` is considered immutable
```

Comment:
Is metatype considered immutable because in the gsi_std input format it's not defined as being signable, or because of some intrinsic property of timestamps, or...?

Reply:
Correct. Clarified.